src/v/cloud_storage/cache_service.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/iostream.hh>
+
+#include <filesystem>
+
+namespace cloud_storage {
+
+struct cache_item {
+    ss::input_stream<char> body;
+    size_t size;
+};
+
+class cache {
+public:
+    /// C-tor.
+    ///
+    /// \param cache_dir is a directory where cached data is stored
+    /// \param max_cache_size is a maximum size of the cached data on disk
+    /// \param cache_period is a time interval used to check if cache should be
+    /// cleaned
+    cache(
+      std::filesystem::path cache_dir,
+      size_t max_cache_size,
+      ss::lowres_clock::duration check_period) noexcept;
+
+    ss::future<> start();
+    ss::future<> stop();
+
+    /// Get cached value as a stream if it exists on disk
+    ss::future<std::optional<cache_item>> get(std::filesystem::path key);
+
+    /// Add new value to the cache, overwrite if it's already exist
+    ss::future<> put(std::filesystem::path key, ss::input_stream<char> data);
+
+    /// Return true if the following object is already in the cache
+    ss::future<bool> is_cached(const std::filesystem::path& key);
+
+    /// Remove element from cache by key
+    ss::future<> invalidate(const std::filesystem::path& key);
+
+private:
+    std::filesystem::path _cache_dir;
+    size_t _max_cache_size;
+    ss::lowres_clock::duration _check_period;
```

Comment:
are you planning on using _check_period in this PR? it would be preferable to either implement it or remove it until the feature is added. i think same applies to max_cache_size.

Reply:
fixed.